packages/typespec-azure-resource-manager/lib/common-types/types.tsp

```diff
@@ -281,172 +243,339 @@ union Origin {
-@armCommonDefinition("CheckNameAvailabilityResponse", Azure.ResourceManager.CommonTypes.Versions.v5)
-@doc("The check availability result.")
 model CheckNameAvailabilityResponse {
-  @doc("Indicates if the resource name is available.")
+  /** Indicates if the resource name is available. */
   nameAvailable?: boolean;
 
-  @doc("The reason why the given name is not available.")
+  /** The reason why the given name is not available. */
   reason?: CheckNameAvailabilityReason;
 
-  @doc("Detailed reason why the given name is not available.")
+  /** Detailed reason why the given name is not available. */
   message?: string;
 }
 
 /**
  * Possible reasons for a name not being available.
  */
-@doc("Possible reasons for a name not being available.")
 union CheckNameAvailabilityReason {
-  @doc("Name is invalid.")
+  /** Name is invalid. */
   Invalid: "Invalid",
 
-  @doc("Name already exists.")
+  /** Name already exists. */
   AlreadyExists: "AlreadyExists",
 
   string,
 }
+
+union EncryptionStatus {
+  enabled: "enabled",
+  disabled: "disabled",
+  string,
+}
+/**
+ * Configuration of key for data encryption
+ */
+model EncryptionProperties {
+  /** Indicates whether or not the encryption is enabled for container registry. */
+  status?: EncryptionStatus;
+
+  /** Key vault properties. */
+  keyVaultProperties?: KeyVaultProperties;
+}
+
+/**
+ * Metadata pertaining to the geographic location of the resource.
+ */
+model LocationData {
+  /** A canonical name for the geographic or physical location. */
+  @maxLength(256)
+  name: string;
+
+  /** The city or locality where the resource is located. */
+  city?: string;
+
+  /** The district, state, or province where the resource is located. */
+  district?: string;
+
+  /** The country or region where the resource is located */
+  countryOrRegion?: string;
+}
+
+/**
+ * The default api-version parameter type.
+ */
+@doc("The default api-version parameter type.")
+model ApiVersionParameter {
+  @query("api-version")
+  @minLength(1)
+  @doc("The API version to use for this operation.")
+  apiVersion: string;
+}
+
+/**
+ * The default operationId parameter type.
+ */
+model OperationIdParameter {
+  /** The ID of an ongoing async operation. */
+  @path
+  @minLength(1)
+  operationId: string;
+}
+
+/**
+ * The default location parameter type.
+ */
+model LocationParameter {
+  /** The name of Azure region. */
+  @path
+  @minLength(1)
+  @segment("locations")
+  location: string;
+}
+
+/**
+ * The default resource group parameter type.
+ */
+model ResourceGroupNameParameter {
+  /** The name of the resource group. The name is case insensitive. */
+  @path
+  @minLength(1)
+  @maxLength(90)
+  @segment("resourceGroups")
+  @pattern("^[-\\w\\._\\(\\)]+$")
+  resourceGroupName: string;
+}
+
+/**
+ * The default subscriptionId parameter type.
+ */
+@doc("The default subscriptionId parameter type.")
+model SubscriptionIdParameter {
```

Comment:
The description also changed from v4. For the versions before v4, it is "The ID of the target subscription." without "The value must be an UUID."

Reply:
The description emitted is the property @doc. Unfortunately, there is no easy way to version the decorators unless you add and remove the properties themselves. I didn't think it is critical here.